sensor/kubernetes/sensor/sensor.go

```diff
@@ -63,7 +62,10 @@ func CreateSensor(cfg *CreateOptions) (*sensor.Sensor, error) {
 		log.Infof("Running sesnor with Kubernetes re-sync enabled. Re-sync time: %s", cfg.resyncPeriod.String())
 	}
 
-	admCtrlSettingsMgr := admissioncontroller.NewSettingsManager(resources.DeploymentStoreSingleton(), resources.PodStoreSingleton())
+	storeProvider := resources.InitializeStore()
+	scan.SetMatchingRegistryFunction(storeProvider.Registries().GetRegistryForImage)
```

Comment:
It makes sense what you did here, but I think it leads to an API that can be a bit confusing. We're injecting a dependency that is used by a public static function (`EnrichLocalImage`), and if we forget to set it we've introduced a panic in:
```go
// sensor/common/scan/scan.go:54
matchingRegistry, err := getMatchingRegistry(ci.GetName())
```

Since you're at this, my suggestion is to convert `sensor/common/scan/scan.go` in a struct and `EnrichLocalImage` in a method of that struct. You can request a `func(*storage.ImageName) (registryTypes.Registry, error)` parameter or the registry store directly. This makes the dependency injection clearer and less error prone. 

Additionally, you can inject this new struct in the only place where it's used, rather than having a public API that is not used anywhere: `sensor/common/image/service_impl.go`. This service already has a dependency on registry store, so if you want to initialize the dependency in the service constructor that's also possible.

Reply:
That's a good suggestion. I've implemented the changes creating a `Scan` struct and wrapping all those functions in it. Let me know what you think.